src/loaders/common/callback_server_wrapper.py

```diff
@@ -99,9 +121,11 @@ def _setup_callback_server_envs(
 
         # used by the callback server
         env["KB_AUTH_TOKEN"] = token
+        env["KB_ADMIN_AUTH_TOKEN"] = token  # pass in admin_token to get catalog params
```

Comment:
It needs to be documented at any API or CLI interfaces that accept the token if and when it needs to be a catalog admin token

Reply:
You mean add a CLI arg `KB_ADMIN_AUTH_TOKEN`?

Reply:
Updated!

Reply:
I'm not seeing any docs in the downloader or uploader about this, am I missing something?

Reply:
I added it here: https://github.com/kbase/collections/blob/9ba9a5504b98628c6dd02cfd570ebc6a7cce985f/src/loaders/common/callback_server_wrapper.py#L111

Reply:
Right, but the actual users of the scripts will never see that. It needs to be documented in the CLIs

Reply:
Also that's on a private method. It should at least be documented at the point where it enters the class, which is the constructor

Reply:
> Also used as a catalog admin token if valid.

What happens with the uploader if it's not a valid admin token?

Reply:
<img width="1115" alt="Screen Shot 2023-12-14 at 10 20 45 PM" src="https://github.com/kbase/collections/assets/22202096/cf9def52-8450-40dc-951b-480cdbaa4412">

Reply:
That's just an invalid token. I'm talking about the case where the token is valid but isn't a catalog admin

Reply:
<img width="1105" alt="Screen Shot 2023-12-18 at 6 29 11 PM" src="https://github.com/kbase/collections/assets/22202096/eb7028d6-bf73-491e-9cbe-51ed0558b9e2">

Reply:
Ok, so wherever the token is required to be a catalog admin token it needs to be documented as such - or it needs to check the token type and skip whatever it's doing that causes the error.
Really if it's required the first thing that should happen is that we check the token and give a better error message - that's pretty cryptic for someone that doesn't understand how all this stuff fits together.

Reply:
Updated. Documented in CLI, cbs constructor, and comment

Reply:
Will the downloader work if the token isn't a catalog admin?

Reply:
Also it should be documented in the downloader that the token is required, and if it's not provided in the file arg  env var {varname} will be checked

Reply:
I still don't see any docs in the uploader CLI...?

Reply:
> Will the downloader work if the token isn't a catalog admin?

I probably need to add a if condition since a catalog admin token is only needed for the uploader.

Reply:
> I still don't see any docs in the uploader CLI...
https://github.com/kbase/collections/blob/2e01b64bbf4adc3dc5fab6a298a1290d29f17114/src/loaders/workspace_uploader/workspace_uploader.py#L132

Reply:
Oh, I was looking here: https://github.com/kbase/collections/blob/2e01b64bbf4adc3dc5fab6a298a1290d29f17114/src/loaders/workspace_uploader/workspace_uploader.py#L27-L29

Reply:
That being said, the line you link is a little inaccurate - for the uploader, the token *must* be an admin token or the upload will fail (and same for the downloader in the current PR code)

Reply:
> the token must be an admin token

It makes me feel it must be a different token. But indeed is just a normal `auth_token` granted admin privilege.

Reply:
maybe the way this should work is there's a --as_catalog_admin CLI param and only if that's true (default false) does the env var get set for the catalog token. Then it's more clear to the users what's going on and non-catalog admins can still run the *loaders (although they'll be stuck with the default catalog secure params)

Reply:
Can you add the language describing that an auth token is required & alternatively can be supplied in an env var to the downloader CLI please?

Reply:
👍

Reply:
:+1: